go/arrow/array/encoded.go

```diff
@@ -68,6 +70,102 @@ func (r *RunEndEncoded) Release() {
 	r.ends.Release()
 }
 
+// LogicalValuesArray returns an array holding the values of each
+// run, only over the range of run values inside the logical offset/length
+// range of the parent array.
+//
+// Example
+//
+// For this array:
+//     RunEndEncoded: { Offset: 150, Length: 1500 }
+//         RunEnds: [ 1, 2, 4, 6, 10, 1000, 1750, 2000 ]
+//         Values:  [ "a", "b", "c", "d", "e", "f", "g", "h" ]
+//
+// LogicalValuesArray will return the following array:
```

Comment:
The C++ implementation has an iterator class that allows a zero-copy iteration over the runs that match the logical `(offset, length)` slice. Do you have a similar thing in Go or is this function used instead?

Reply:
Got it.